libcontainerd/container_linux.go

```diff
@@ -145,11 +153,14 @@ func (ctr *container) handleEvent(e *containerd.Event) error {
 
 		// Remove process from list if we have exited
 		// We need to do so here in case the Message Handler decides to restart it.
-		if st.State == StateExit {
+		switch st.State {
+		case StateExit:
 			if os.Getenv("LIBCONTAINERD_NOCLEAN") != "1" {
 				ctr.clean()
 			}
 			ctr.client.deleteContainer(e.Id)
+		case StateExitProcess:
+			ctr.cleanProcess(st.ProcessID)
```

Comment:
I didn't want to change that atm because the debug flag has some issues atm when containers exit with error before starting. Also, the flag is for debugging config/rootfs, the fifos are unusable anyway after process has exited. So maybe we should remove this flag or provide a better implementation but it can be in a different change. I also just didn't want to simply remove it because it may still provide useful information in certain cases.